### PR TITLE
Expose stream-ordering in nvtext API

### PR DESCRIPTION
**Pull request by shrshi**
## Description
Adds stream parameter to
```
cudf::nvtext::byte_pair_encoding
```
Added stream gtests to verify correct stream forwarding.

Reference: https://github.com/rapidsai/cudf/issues/13744

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/streams/text/bpe_test.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    nvtext::load_merge_pairs(cudf::strings_column_view(mpt), cudf::test::get_default_stream());
+
+  auto validity = cudf::test::iterators::null_at(4);
+  cudf::test::strings_column_wrapper input(
+    {"thisisit", "thisis test-sentence-1", "thisistestsentence-2", "this-istestsentence 3", "", ""},
+    validity);
+  auto sv = cudf::strings_column_view(input);
+
+  auto results = nvtext::byte_pair_encoding(
+    sv, *merge_pairs, cudf::string_scalar(" "), cudf::test::get_default_stream());
```
Comment (davidwendt) [inline cpp/tests/streams/text/bpe_test.cpp:58]:
```suggestion
    sv, *merge_pairs, cudf::string_scalar(" ", true, cudf::test::get_default_stream()), cudf::test::get_default_stream());
```
Maybe you can also create `auto stream = cudf::test::get_default_stream();` to use here.

Comment (davidwendt) [reply cpp/tests/streams/text/bpe_test.cpp:58]: You will also need to make the same change to the scalar used here:
https://github.com/rapidsai/cudf/blob/953a5d5ae3cfd82f03fe394fd75d0c1bb3bcbeea/cpp/src/text/bpe/load_merge_pairs.cu#L106

Comment (shrshi) [reply cpp/tests/streams/text/bpe_test.cpp]: Thanks, David! That fixed it :)

**Review by PointKernel — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Pull request merged**